src/mscorlib/src/System/Threading/Mutex.cs

```diff
@@ -52,17 +52,17 @@ public Mutex(bool initiallyOwned, String name, out bool createdNew)
         [ReliabilityContract(Consistency.WillNotCorruptState, Cer.MayFail)]
         public unsafe Mutex(bool initiallyOwned, String name, out bool createdNew, MutexSecurity mutexSecurity)
         {
-            if (name != null)
+            if (name == string.Empty)
             {
-#if PLATFORM_UNIX
-                throw new PlatformNotSupportedException(Environment.GetResourceString("PlatformNotSupported_NamedSynchronizationPrimitives"));
-#else
-                if (System.IO.Path.MaxPath < name.Length)
-                {
-                    throw new ArgumentException(Environment.GetResourceString("Argument_WaitHandleNameTooLong", name));
-                }
-#endif
+                // Empty name is treated as an unnamed mutex. Set to null, and we will check for null from now on.
+                name = null;
             }
+#if !PLATFORM_UNIX
+            if (name != null && System.IO.Path.MaxPath < name.Length)
```

Comment:
Is there a length limitation on Unix, and if so, do we need to check for that here, too?